pkg/engine/virtualmachines.go

```diff
@@ -498,7 +498,25 @@ func createAgentAvailabilitySetVM(cs *api.ContainerService, profile *api.AgentPo
 	storageProfile := compute.StorageProfile{}
 
 	if profile.IsWindows() {
-		storageProfile.ImageReference = createWindowsImageReference(profile.Name, cs.Properties.WindowsProfile)
```

Comment:
Should we add these into `createWindowsImageReference` instead?

Reply:
I planned to remove createWindowsImageReference  later. I think that keeping the logic here as Linux does makes it more readable.

Reply:
But it leaves `createWindowsImageReference` in a wrong logic, I prefer we leave a TODO comment to fix it later.  It depends on you.